cmd/srcd/cmd/sql.go

```diff
@@ -33,37 +33,75 @@ var sqlCmd = &cobra.Command{
 	Use:   "sql",
 	Short: "Run a SQL query over the analyzed repositories.",
 	RunE: func(cmd *cobra.Command, args []string) error {
-		if len(args) < 1 {
-			return fmt.Errorf("missing query argument")
-		} else if len(args) > 1 {
-			return fmt.Errorf("two many arguments, expected only one query")
+		if len(args) > 1 {
+			return fmt.Errorf("two many arguments, expected only one query or nothing")
 		}
-		query := args[0]
 
-		c, err := daemon.Client()
-		if err != nil {
-			logrus.Fatalf("could not get daemon client: %v", err)
+		var query string
+		if len(args) == 1 {
+			query = args[0]
+		}
+
+		if strings.TrimSpace(query) == "" {
+			return repl()
 		}
 
-		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
-		defer cancel()
+		return runQuery(query)
+	},
+}
+
+func repl() error {
+	rl, err := readline.NewEx(&readline.Config{
+		Prompt: "gitbase> ",
+		Stdin:  os.Stdin,
+		Stderr: os.Stderr,
+		Stdout: os.Stdout,
+	})
+	if err != nil {
+		return err
+	}
+	defer rl.Close()
 
-		res, err := c.SQL(ctx, &api.SQLRequest{Query: query})
+	for {
+		line, err := rl.Readline()
 		if err != nil {
-			logrus.Fatalf("server error: %v", err)
+			return nil
 		}
 
-		w := new(tabwriter.Writer)
-		w.Init(os.Stdout, 0, 8, 2, '\t', 0)
-		defer w.Flush()
-
-		fmt.Fprintln(w, strings.ToUpper(strings.Join(res.Header.Cell, "\t")))
-		fmt.Fprintln(w, "-------------")
-		for _, row := range res.Rows {
-			fmt.Fprintln(w, strings.Join(row.Cell, "\t"))
+		switch strings.ToLower(strings.TrimSpace(line)) {
+		case "exit", "quit":
+			return nil
+		default:
+			if err := runQuery(line); err != nil {
+				fmt.Println(err)
+			}
 		}
-		return nil
-	},
+	}
+}
+
+func runQuery(query string) error {
+	c, err := daemon.Client()
+	if err != nil {
+		return fmt.Errorf("could not get daemon client: %v", err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	res, err := c.SQL(ctx, &api.SQLRequest{Query: query})
+	if err != nil {
+		return err
```

Comment:
I'd consider adding at least at `TODO` in here regarding the error messages.
Whenever a syntax error occurs in the backend the error message looks like:

```
rpc error: code = Unknown desc = SQL query failed: Error 1105: unknown error: syntax error at position 5 near 'exit'
```

I think it'd be better to identify this was not a transport error somehow and print just the `syntax error at position 5 near 'exit'`